pkg/apis/cluster/validation/validation.go

```diff
@@ -38,8 +38,7 @@ func ValidateClusterName(name string) []string {
 }
 
 var (
-	supportedSyncModes     = sets.NewString(string(api.Pull), string(api.Push))
-	supportedResourceNames = sets.NewString(string(api.ResourceCPU), string(api.ResourceMemory), string(api.ResourceStorage), string(api.ResourceEphemeralStorage))
```

Comment:
I understand that the value of user resources should be planned in advance and not too arbitrary. Can we set the key values of supported resources for users from a configurable perspective, while still retaining the validation behavior?

Reply:
how about a env of karmada-apiserver？I don't want to add any new api for this, a launch param or env is acceptable

Reply:
I personally prefer to start with parameter or configuration files or resources.

Reply:
I think it is no big influence without validation, if user mistake in writing in any one grade, we have other ways to report error (the resource's range should be defined in every grade, others report error)

so, I advice to keep unchanged for the moment.